src/WebJobs.Script/Description/FunctionInvokerBase.cs

```diff
@@ -28,21 +27,42 @@ internal FunctionInvokerBase(ScriptHost host, FunctionMetadata functionMetadata,
         {
             Host = host;
             Metadata = functionMetadata;
+            _metrics = host.ScriptConfig.HostConfig.GetService<IMetricsLogger>();
 
+            // Function file logging is only done conditionally
             traceWriterFactory = traceWriterFactory ?? new FunctionTraceWriterFactory(functionMetadata.Name, Host.ScriptConfig);
             TraceWriter traceWriter = traceWriterFactory.Create();
+            FileTraceWriter = traceWriter.Conditional(t => Host.FileLoggingEnabled && (!(t.Properties?.ContainsKey(PrimaryHostTracePropertyName) ?? false) || Host.IsPrimary));
 
-            // Function file logging is only done conditionally
-            TraceWriter = traceWriter.Conditional(t => Host.FileLoggingEnabled && (!(t.Properties?.ContainsKey(PrimaryHostTracePropertyName) ?? false) || Host.IsPrimary));
-            _metrics = host.ScriptConfig.HostConfig.GetService<IMetricsLogger>();
+            // The global trace writer used by the invoker will write all traces to both
+            // the host trace writer as well as our file trace writer
+            TraceWriter = host.TraceWriter != null ? 
+                new CompositeTraceWriter(new TraceWriter[] { FileTraceWriter, host.TraceWriter }) : 
+                FileTraceWriter;
+
+            // Include the function name in the properties so system logs
+            // will include it
+            PrimaryHostTraceProperties = new Dictionary<string, object>
+            {
+                { PrimaryHostTracePropertyName, null },
+                { ScriptConstants.AzureFunctionsTracePropertyFunctionNameKey, Metadata.Name }
```

Comment:
Agree with this, perhaps you could add the function name as a property to the FileTraceWriter from the FunctionTraceWriterFactory where it's already passed in, but there might be complexity in handling this with CompositeTraceWriter.

Alternatively I'm sure there's a way to wrap / decorate the tracewriter so that it supplies the function name as a property.